pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -925,10 +925,22 @@ self: super: {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
   };
-  Agda = super.Agda.override {
+  Agda = overrideCabal (super.Agda.override {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
-  };
+  }) (drv: {
+    postInstall = ''
+      # Separate loops to avoid internal error
+      files=($out/share/*-ghc-*/Agda-*/lib/prim/Agda/{Primitive.agda,Builtin/*.agda})
+      for f in "''${files[@]}" ; do
+        $out/bin/agda $f
+      done
+      for f in "''${files[@]}" ; do
+        $out/bin/agda -c --no-main $f
+      done
+      $out/bin/agda-mode compile
+    '';
+  });
 
```

Comment:
I somewhat agree. The `postInstall` in `cabal2nix` is still okay for <2.5 versions, so it should be preserved somewhere. I guess doing a similar override on `Agda_2_4_2_3` and `Agda_2_4_2_4` in `configuration-common.nix` (and getting rid of the stuff in `cabal2nix`) would be a reasonably sensible thing to do.